paddle/fluid/operators/flatten_op.cc

```diff
@@ -157,6 +157,105 @@ class FlattenGradOp : public framework::OperatorBase {
   }
 };
 
+// FIXME(zcd): flatten2 adds an intermediate output(XShape) based on flatten,
+// the XShape is used to carry the shape and lod of X which will be used in
+// flatten_grad, in this way, the framework can reuse the memory of X
+// immediately the flatten2_op is finished.
+// Considering compatibility issues, we could not fix flatten2_op
+class Flatten2OpInferShape : public FlattenOpInferShape {
+ public:
+  void operator()(framework::InferShapeContext *ctx) const override {
+    FlattenOpInferShape flatten_op_infer_shape;
+    flatten_op_infer_shape(ctx);
```

Comment:
You means `FlattenOpInferShape::operator()(ctx)`?

Reply:
Yes, I have fixed it.